### DISCUSSION TITLE
Geneach / Await (all agent code) broken since 0.0.62, please help?

### DISCUSSION THREAD

**Discussion by TheSylvester** [guidance-ai/guidance#293] [Q&A]
Multiple issues have been opened for the same bug - geneach / await is broken, and either won't keep old items in the geneach list for the next call, or doesn't pick up the user's input variables to the template:

https://github.com/microsoft/guidance/issues/260
https://github.com/microsoft/guidance/issues/262
https://github.com/microsoft/guidance/issues/278

The previous solution was to lock the version at guidance==0.0.61, but either the way the LLM is responding or some other underlying dependency has changed, and even our 0.0.61 implementation stopped working as of yesterday.

From experimentation, this strange behaviour may have something to do with async operations and the loop.  Oddly, even changing the name of the response variable from `prompt = guidance(....)` and `prompt = prompt(....)` has an effect;  Putting those statements at top-level has an effect;  Executing the same statements as part of a function causes even worse crashes.

Could someone else independently verify this?  The test_program.py unit test seems unable to cover this case and passes with the buggy code, as it has the perfect variable names and is only testing for whether the LLM is giving an output, which it is, just not one that takes the user input into account.

Assistance would be much appreciated as this is a breaking bug that has gone on for weeks now

**Comment by TheSylvester** [comment]
I strongly suspect this is all part of Partials being completely broken and await functionality gone:

https://github.com/microsoft/guidance/issues/296

**Comment by fullstackwebdev** [comment]
Did you ever find a solution?

**Comment by slundberg** [comment]
This is something that was a result of having to pause dev work during the summer. With v0.1 you now no longer even need to use `await` because there is no longer a distinction between user code and template code, (it is all just native Python). So I am going to close this for now. If you to write an agent you can keep track of state yourself in native Python.
